coffeecard/CoffeeCard.Library/Services/v2/ProductService.cs

```diff
@@ -57,6 +58,111 @@ public async Task<Product> GetProductAsync(int productId)
             return product;
         }
 
+        public Task<InitiateProductResponse> AddProduct(InitiateProductRequest newProduct, IEnumerable<UserGroup> allowedUserGroups)
+        {
+            var product = new Product()
+            {
+                Price = newProduct.Price,
+                Description = newProduct.Description,
+                Name = newProduct.Name,
+                NumberOfTickets = newProduct.NumberOfTickets,
+                Visible = newProduct.Visible
+            };
+            
+            var productUserGroups = allowedUserGroups.Select(userGroup => new ProductUserGroup
+            {
+                ProductId = product.Id,
```

Comment:
Have you tested product have an Id assigned at this point? It has not been added to the db yet, so likely has no ID

Reply:
Alternatively, you can just reference the product object and Entity Framework will do the stitching itself